tests/index_test.py

```diff
@@ -0,0 +1,218 @@
+            }]
+        }]
+        def get_past_time(hours: int=0) -> str:
+            return (
+                datetime.now(timezone.utc) - timedelta(hours=hours)).strftime(
+                    "%Y-%m-%dT%H:%M:%SZ")
+        self.timeline1 = [{
+            'event': 'created'
+        }, {
+            'event': 'assigned',
+            'assignee': {
+                'login': 'reviewerName1'
+            },
+            'created_at': get_past_time(hours=22)
+        },{
+            'event': 'assigned',
+            'assignee': {
+                'login': 'reviewerName2'
+            },
+            'created_at': get_past_time(hours=56)
+        }]
+
+        self.timeline2 = [{
+            'event': 'created'
+        }, {
+            'event': 'assigned',
+            'assignee': {
+                'login': 'reviewerName1'
+            },
+            'created_at': get_past_time(hours=23)
+        }, {
+            'event': 'assigned',
+            'assignee': {
+                'login': 'reviewerName2'
+            },
+            'created_at': get_past_time(hours=19)
+        }]
+        self.test_template = "{{ username }}\n{{ pr_list }}"
+
+    def mock_all_get_requests(self, mock_request: requests_mock) -> None:
+        param_page_1='?page=1&per_page=100'
+        param_page_2='?page=2&per_page=100'
+        mock_request.get(
+            github_services.PULL_REQUESTS_URL_TEMPLATE.format(
+                self.orgName, self.repoName) + param_page_1,
+            text=json.dumps(self.pull_response))
+        mock_request.get(
+            github_services.PULL_REQUESTS_URL_TEMPLATE.format(
+                self.orgName, self.repoName) + param_page_2,
+            text=json.dumps([]))
+
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.orgName, self.repoName, 123) + param_page_1,
+            text=json.dumps(self.timeline1))
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.orgName, self.repoName, 123) + param_page_2,
+            text=json.dumps([]))
+
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.orgName, self.repoName, 234) + param_page_1,
+            text=json.dumps(self.timeline2))
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.orgName, self.repoName, 234) + param_page_2,
+            text=json.dumps([]))
+
+    def test_executing_main_function_sends_notification(self) -> None:
+        with requests_mock.Mocker() as mock_request:
+            self.mock_all_get_requests(mock_request)
+                
+            mock_resp_1 = Mock()
+            mock_resp_1.json.return_value = self.response_for_discussions
+
+            mock_resp_2 = Mock()
+            mock_resp_2.json.return_value = self.response_for_comment
+            with patch("requests.post", side_effect=[
```

Comment:
Use single quotes here and below.

Also, I'm having a bit of trouble understanding this test. I suggest adding some comments explaining conceptually what you are trying to do.

Reply:
Replaced double quotes and added comments explaining the steps